agora/Agora/Effect/GovernorMutation.hs

```diff
@@ -66,8 +68,8 @@ import "liqwid-plutarch-extra" Plutarch.Extra.TermCont (pguardC, pletC, pletFiel
      @since 0.1.0
 -}
 data MutateGovernorDatum = MutateGovernorDatum
-  { governorRef :: TxOutRef
-  -- ^ Referenced governor state UTXO should be updated by the effect.
+  { oldDatum :: GovernorDatum
+  -- ^ The governor datum value on which this effect is valid
```

Comment:
I still don't think pining the governor datum is a good idea - as I discussed with @t4ccer earlier, a malicious user can always stop the effect from executing by simply making a proposal.

> Adding a nonce to the governor datum that is incremented on each mutation, and restricting based on this

This requires changing the governor script. We don't want to do this, at least not before agora v2. cc @emiflake 

> It may be worth adding functionality to cancel an effect that can't be applied (either because the datum doesn't match or the effect has expired)

Good idea, the key problem is that we can't safely purge a GAT atm.

Reply:
> Restricting the effect to a fixed time window

We might want to go this route. We can have a field in the datum which specifies in what time range the GAT is allowed to be burnt.

Reply:
I don't like adding time range because:
- It locks GAT token after deadline, which is risky if there's a bug in the effect script
- You can still do some malicious stuff after the timeframe

I think that `proposalId` should never be modified by the effect, and we can then add nonce acting as "datum version" to make sure updates are executed in order

Reply:
> It locks GAT token after deadline, which is risky if there's a bug in the effect script

We can make it so that the effect script ensures that the GAT can be burnt after the deadline, but the actual effect cannot be executed. This does add more points of failure, though. 

> Could makeEffect just provide the effect function with either a RunEffect or CancelEffect value and have the effect determine its own conditions for each? The check for the GAT being burned can apply in either case.

Unless adding an authcheck script, you have no way to signal the effect script cause it doesn't (and shouldn't) care about the redeemer. This behaviour is in the agora spec.